Explorer/Assets/Scripts/SceneRunner/Scene/SceneContentProvider.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SceneRunner.Scene
+{
+    public class SceneContentProvider : ISceneContentProvider
+    {
+        private readonly Dictionary<string, string> fileToHash;
+
+        private readonly ISceneData sceneData;
+        private readonly bool supportHashes;
+
+        /// <param name="sceneData"></param>
+        /// <param name="supportHashes">
+        ///     Indicates if hashes should be used for resources loading.
+        ///     To load from local files it should be set to false
+        /// </param>
+        public SceneContentProvider(ISceneData sceneData, bool supportHashes)
+        {
+            this.sceneData = sceneData;
+            this.supportHashes = supportHashes;
+
+            if (!supportHashes) return;
+
+            fileToHash = new Dictionary<string, string>(sceneData.Contents.Count, StringComparer.OrdinalIgnoreCase);
+
+            for (var i = 0; i < sceneData.Contents.Count; i++)
+            {
+                ISceneData.ContentMappingPair mapping = sceneData.Contents[i];
+                fileToHash[mapping.file] = mapping.hash;
+            }
+        }
+
+        public string SceneName => sceneData.Name;
+
+        public bool HasRequiredPermission(string permission)
+        {
+            for (var i = 0; i < sceneData.RequiredPermissions.Count; i++)
+            {
+                if (sceneData.RequiredPermissions[i] == permission)
+                    return true;
+            }
+
+            return false;
+        }
+
+        public bool TryGetContentUrl(string url, out string result)
+        {
+            if (supportHashes)
+            {
+                if (fileToHash.TryGetValue(url, out string hash))
+                {
+                    result = sceneData.BaseUrl + hash;
+                    return true;
+                }
+
+                Debug.LogError($"{nameof(SceneContentProvider)}: {url} not found in {nameof(fileToHash)}");
```

Comment:
We should discuss how logging should be handled in the PoC - one major issue now is that we have nothing to distinguish between user error (from the scenes themselves) and errors that we should fix. Leading to millions of errors! Ideally for performance we should not even have logging at runtime?

Reply:
Let's discuss